src/hist/intervals.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Optional
+
+import numpy as np
+from scipy import stats
+
+__all__ = ("poisson_interval", "clopper_pearson_interval", "ratio_uncertainty")
+
+
+# Code originally contributed to coffea by Nicholas Smith
+# Copyright (c) 2018, Fermilab
```

Comment:
@nsmith- If you want to review the intervals module that'd be cool too. :+1:

Reply:
lmk if you want it here or in a new pr

Reply:
@nsmith- if you've got feedback now feel free to unload it here and I can migrate any discussion points to the new PR.

Reply:
I haven't looked at it, seems easier to do it on the new one if its made, or here if not.

Reply:
SGTM.

Reply:
Allow me to chime in - is this copyright statement not going to "blow you in the face" down the line? First off, why would it not simply be copyright Nicholas Smith and why Fermilab, really? Would it not be better/safer to simply acknowledge the code from ... It's not been a week since I heard all kinds of worries related to copyrights, so ... And you're sure that this is OK with respect to BSD-3? Again, better play it safe and only complicate if no other reason, and for a real benefit. My 2 cents. Sorry for the interjection.

Reply:
coffea is BSD-3 as well (which is allowed for fnal stuff), so the code can be lifted with attribution and inclusion of the license. What's not clear is how to attribute and include the license. @lgray may have ideas

Reply:
Ah, that's nicer / good to know. I'm not sure either. But scikit-learn has plenty of code authored by many, explicitly mentioned in the files, and that does not cause problems for the license itself, which is a single one for "the developers". I would have to dive a bit more to see anything subtle in there.

Reply:
At a minimum you just need to include a URL for the license in coffea so you prove the code base is under it.
I'd just take the URL of the license in the coffea github repository and stick it here, should be good enough and then no possibility of blowing up in your face.

Reply:
As far as specificity of inclusion I think it's nominally to the entity in the license in the original repo. So "Fermilab" is the copyright holder.  You can probably include names of authors for fun.

Reply:
I believe if you have the permissions of all the original authors, they can relicense it here, simliar to if they had just directly contributed it.

Reply:
Yeah, the BSD3 nature of the coffea, specifically the 

> * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
  
  part is what prompted this (and you're right @lgray, the inclusion of Nick's name was just to give him a hat tip).
  
> I believe if you have the permissions of all the original authors, they can relicense it here, simliar to if they had just directly contributed it.

Ah this is very good to know! As I'm going to factor this out in the service of more atomic PRs let me ping all of you again on this on the new PR in a bit (unless @lgray and @nsmith- you agree with Henry here and would prefer to not get further pings on this).

Reply:
This is now PR #176